src/pages/index.js

```diff
@@ -4,27 +4,113 @@ import "../pages/index.css";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
+import PopupConfirmDelete from "../components/ConfirmDelete.js";
+import PopupWithProfileImage from "../components/PopupWithProfileImage.js";
+import Api from "../components/Api.js";
 import UserInfo from "../components/UserInfo.js";
 import { initialCards, options } from "../utils/constants";
 
 // Initialize user information
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__job",
+  profileImageSelector: ".profile__image",
 });
 
 // Form elements and validators
 const profileFormElement = document.forms["modal__profile-form"];
 const cardFormElement = document.forms["modal__card-form"];
 const profileFormValidator = new FormValidator(options, profileFormElement);
 const cardFormValidator = new FormValidator(options, cardFormElement);
+
+const deleteCardFormElement = document.forms["modal__delete-card-form"];
+const editProfileFormElement = document.forms["modal__profile-image"];
+const deleteCardFormValidator = new FormValidator(
+  options,
+  deleteCardFormElement
+);
+const editProfileFormValidator = new FormValidator(
+  options,
+  editProfileFormElement
+);
+
 const popupWithImage = new PopupWithImage("#modal_image");
+const userImage = document.querySelector(".profile__image");
+
+// Initialize the profile image modal
+const popupUpdateProfileImage = new PopupWithProfileImage({
+  popupSelector: "#modal_profile-image",
+  imageSelector: ".profile__image-container",
+  handleFormSubmit: (formData) => {
+    return api
+      .updatingProfile({ profileImage: formData.profileImage })
+      .then((updatedUserData) => {
+        userInfo.setUserInfo(updatedUserData);
+      })
+      .catch((err) => {
+        console.error(`Failed to update profile image: ${err}`);
+      });
+  },
+});
+
+popupUpdateProfileImage._setClickListener();
+
+const popupConfirmDeletes = new PopupConfirmDelete({
+  popupSelector: "#modal_delete-card",
+  handleDeleteSubmit: (card) => {
+    api
+      .deleteCard(card.id)
+      .then(() => {
+        card.element.remove();
+        popupConfirmDeletes.close();
+      })
+      .catch((err) => {
+        console.error(`Failed to delete the card: ${err}`);
+      });
+  },
+}); //??????????????????????????????
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "95098c46-93e7-4f0b-a6fc-0aca47be17cd",
+    "Content-Type": "application/json",
+  },
+});
+// uses api to get initial cards
+
+api
+  .getInitialCards()
+  .then((result) => {
+    section.renderItems(result);
+  })
+  .catch((err) => {
+    console.error(err); // log the error to the console
+  });
+
+///////
+
+// Fetch user info and set it on the page
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo({ name: userData.name, job: userData.about });
+    userImage.src = userData.avatar;
```

Comment:
`userImage.src = userData.avatar;`

You should use the `setUserInfo` for setting the `avatar` as well


```js
api
  .getUserInfo()
  .then((userData) => {
    userInfo.setUserInfo({ name: userData.name, job: userData.about, avatar: userData.avatar });  // here
  })
  .catch((err) => {
    console.error(`Failed to fetch user data: ${err}`);
  });

```